### PR TITLE
show interface portchannel support for Multi ASIC

### PR DESCRIPTION
**Pull request by arlakshm**
Add the multi ASIC specific options for the command show interface portchannel

Depends on the following PRs
 - [sonic-buildimage/4973](https://github.com/Azure/sonic-buildimage/pull/4973)
 - [sonic-utilities/999](https://github.com/Azure/sonic-utilities/pull/999) 

Signed-off-by: Arvindsrinivasan Lakshmi Narasimhan <arlakshm@microsoft.com>

**- What I did**
Add support for multi ASIC CLI options for `show interface portchannels`
2 new options have added

- [-n, --namespace] to allow user to display the information for given namespaces
  If this option is not present the information from all the namespaces will be displayed

- [-d, --display] to allow user to display information related both internal and external interfaces
  If this option is not present only external interfaces/neighbors will be display

One single ASIC platform, this options are not valid, so the behavior remains unchanged

**- How I did it**
  - Change the teamshow script to get the port channel information from all the Namespaces
  -  Add support for `-n` and `-d` options
This new options are ignored for single ASIC platforms

**- How to verify it**
**Help menu on multi ASIC platforms**
```

admin@sonic:~$ show interface portchannel -h
Usage: show interface portchannel [OPTIONS]

  Show PortChannel information

Options:
  -d, --display [all|frontend]    Show internal interfacees  [default:
                                  frontend]
  -n, --namespace [asic0|asic1|asic2|asic3|asic4|asic5]
                                  Namespace name or all
  --verbose                       Enable verbose output
  -?, -h, --help                  Show this message and exit.
admin@sonic:~$ 
```
**Sample output for show interface portchannel**
```
admin@sonic:~$ show interface portchannel 
Flags: A - active, I - inactive, Up - up, Dw - Down, N/A - not available,
       S - selected, D - deselected, * - not synced
  No.  Team Dev         Protocol     Ports
-----  ---------------  -----------  -----------------------------------------------------------------------------------------------------------------------------------------------
 1001  PortChannel1001  LACP(A)(Up)  Ethernet136(S)
 1002  PortChannel1002  LACP(A)(Up)  Ethernet0(S) Ethernet4(S)
 1003  PortChannel1003  LACP(A)(Up)  Ethernet144(S)
 1004  PortChannel1004  LACP(A)(Up)  Ethernet148(S)
 1005  PortChannel1005  LACP(A)(Up)  Ethernet20(S) Ethernet16(S)
 1006  PortChannel1006  LACP(A)(Up)  Ethernet152(S)
 1007  PortChannel1007  LACP(A)(Up)  Ethernet156(S)
 1008  PortChannel1008  LACP(A)(Up)  Ethernet64(S) Ethernet68(S)
 1009  PortChannel1009  LACP(A)(Up)  Ethernet168(S)
 1010  PortChannel1010  LACP(A)(Up)  Ethernet176(S)
 1011  PortChannel1011  LACP(A)(Up)  Ethernet80(S) Ethernet84(S)
 1012  PortChannel1012  LACP(A)(Up)  Ethernet180(S)
 1013  PortChannel1013  LACP(A)(Up)  Ethernet184(S)
 1014  PortChannel1014  LACP(A)(Up)  Ethernet188(S)
 1015  PortChannel1015  LACP(A)(Up)  Ethernet200(S)
 1016  PortChannel1016  LACP(A)(Up)  Ethernet208(S)
 1017  PortChannel1017  LACP(A)(Up)  Ethernet212(S)
 1018  PortChannel1018  LACP(A)(Up)  Ethernet216(S)
 1019  PortChannel1019  LACP(A)(Up)  Ethernet220(S)
 1020  PortChannel1020  LACP(A)(Up)  Ethernet232(S)
 1021  PortChannel1021  LACP(A)(Up)  Ethernet240(S)
 1022  PortChannel1022  LACP(A)(Up)  Ethernet244(S)
 1023  PortChannel1023  LACP(A)(Up)  Ethernet248(S)
 1024  PortChannel1024  LACP(A)(Up)  Ethernet252(S)
```
**Sample output for single ASIC **
```
admin@sonic:~$ show interface portchannel -n asic0
Flags: A - active, I - inactive, Up - up, Dw - Down, N/A - not available,
       S - selected, D - deselected, * - not synced
  No.  Team Dev         Protocol     Ports
-----  ---------------  -----------  ---------------------------
 1002  PortChannel1002  LACP(A)(Up)  Ethernet0(S) Ethernet4(S)
 1005  PortChannel1005  LACP(A)(Up)  Ethernet20(S) Ethernet16(S)
```
```

admin@sonic:~$ show interface portchannel -n asic0 -d all
Flags: A - active, I - inactive, Up - up, Dw - Down, N/A - not available,
       S - selected, D - deselected, * - not synced
  No.  Team Dev         Protocol     Ports
-----  ---------------  -----------  ---------------------------------------------------------------------------------------------------------------------------------------
 1002  PortChannel1002  LACP(A)(Up)  Ethernet0(S) Ethernet4(S)
 1005  PortChannel1005  LACP(A)(Up)  Ethernet20(S) Ethernet16(S)
 4001  PortChannel4001  LACP(A)(Up)  Ethernet-BP4(S) Ethernet-BP16(S) Ethernet-BP20(S) Ethernet-BP12(S) Ethernet-BP24(S) Ethernet-BP0(S) Ethernet-BP28(S) Ethernet-BP8(S)
 4002  PortChannel4002  LACP(A)(Up)  Ethernet-BP48(S) Ethernet-BP60(S) Ethernet-BP52(S) Ethernet-BP40(S) Ethernet-BP36(S) Ethernet-BP56(S) Ethernet-BP44(S) Ethernet-BP32(S)

```
**- Previous command output (if the output of a command-line utility has changed)**

**- New command output (if the output of a command-line utility has changed)**

### CODE REVIEW

**Comment by jleveque** [comment]
Note that the PR check build will fail until the following PRs are merged (in this order):
1. https://github.com/Azure/sonic-buildimage/pull/5043
2. https://github.com/Azure/sonic-build-tools/pull/141

File: show/main.py [hunk 1]
```diff
@@ -1116,10 +1117,15 @@ def rif(interface, period, verbose):
 
 # 'portchannel' subcommand ("show interfaces portchannel")
 @interfaces.command()
+@multi_asic_click_options
```
Comment (judyjoseph) [inline show/main.py:1120]: For the config commands I add the namespace option at the command group level ( i.e if you take this example ..it will be like show interfaces -n asic0 portchannel/counters ) instead of specifying the options after the subcommand keyword viz portchannel, counters etc ..

I was looking to see if both the show and config commands follow similar approach .. @arlakshm @jleveque your thoughts ?

Comment (arlakshm) [reply show/main.py:1120]: @judyjoseph. IMO the CLI syntax is cleaner if we have the keywords first then followed by options/arguments.

File: show/interfaces/portchannel.py [hunk 2]
```diff
@@ -142,16 +155,14 @@ def display_summary(self):
         output = []
         for team_id in natsorted(self.summary):
             output.append([team_id, 'PortChannel'+team_id, self.summary[team_id]['protocol'], self.summary[team_id]['ports']])
-        print(tabulate(output, header))
+        print tabulate(output, header)
```
Comment (lguohan) [inline show/interfaces/portchannel.py:158]: keep it, make it py3 compatible.

**Comment by lguohan** [comment]
please pass the test

File: show/interfaces/portchannel.py [hunk 3]
```diff
@@ -142,16 +155,14 @@ def display_summary(self):
...
 
 # 'portchannel' subcommand ("show interfaces portchannel")
 @click.command()
+@multi_asic_util.multi_asic_click_options
 @click.option('--verbose', is_flag=True, help="Enable verbose output")
-@clicommon.pass_db
```
Comment (lguohan) [inline show/interfaces/portchannel.py:150]: why is this removed? this can break the unit test.

File: show/interfaces/portchannel.py [hunk 4]
```diff
@@ -35,21 +38,31 @@
...
     def get_portchannel_names(self):
         """
             Get the portchannel names from database.
         """
+        self.teams = []
         team_keys = self.db.keys(self.db.CONFIG_DB, PORT_CHANNEL_CFG_TABLE_PREFIX+"*")
-        if team_keys is None:
-            return
-        self.teams = [key[len(PORT_CHANNEL_CFG_TABLE_PREFIX):] for key in team_keys]
+        for key in team_keys:
```
Comment (lguohan) [inline show/interfaces/portchannel.py:61]: what if team_keys is None? will it throw exception?
